src/Workspaces/Remote/ServiceHub/Services/RemoteHostService.PerformanceReporter.cs

```diff
@@ -0,0 +1,109 @@
+                _reported = new HashSet<string>();
+
+                _logger = logger;
+                _diagnosticAnalyzerPerformanceTracker = diagnosticAnalyzerPerformanceTracker;
+                _diagnosticAnalyzerPerformanceTracker.SnapshotAdded += OnSnapshotAdded;
+                Start();
+            }
+
+            protected override void PauseOnGlobalOperation()
+            {
+                // we won't cancel report already running. we will just prevent
+                // new one from starting.
+            }
+
+            protected override async Task ExecuteAsync()
+            {
+                // wait for global operation such as build
+                await GlobalOperationTask.ConfigureAwait(false);
+
+                using (var pooledObject = SharedPools.Default<List<ExpensiveAnalyzerInfo>>().GetPooledObject())
+                using (RoslynLogger.LogBlock(FunctionId.Diagnostics_GeneratePerformaceReport, CancellationToken))
+                {
+                    _diagnosticAnalyzerPerformanceTracker.GenerateReport(pooledObject.Object);
+
+                    foreach (var badAnalyzerInfo in pooledObject.Object)
+                    {
+                        var newAnalyzer = _reported.Add(badAnalyzerInfo.AnalyzerId);
+                        var internalUser = WatsonReporter.IsUserMicrosoftInternal;
+
+                        // we only report same analyzer once unless it is internal user
+                        if (internalUser || newAnalyzer)
+                        {
+                            // this will report telemetry under VS. this will let us see how accurate our performance tracking is
+                            RoslynLogger.Log(FunctionId.Diagnostics_BadAnalyzer, KeyValueLogMessage.Create(m =>
+                            {
+                                // since it is telemetry, we hash analyzer name if it is not builtin analyzer
+                                m[nameof(badAnalyzerInfo.AnalyzerId)] = internalUser ? badAnalyzerInfo.AnalyzerId : badAnalyzerInfo.PIISafeAnalyzerId;
+                                m[nameof(badAnalyzerInfo.LOF)] = badAnalyzerInfo.LOF;
+                                m[nameof(badAnalyzerInfo.Average)] = badAnalyzerInfo.Average;
+                                m[nameof(badAnalyzerInfo.AdjustedStandardDeviation)] = badAnalyzerInfo.AdjustedStandardDeviation;
+                            }));
+                        }
+
+                        // for logging, we only log once. we log here so that we can ask users to provide this log to us
+                        // when we want to find out VS performance issue that could be caused by analyzer
+                        if (newAnalyzer)
+                        {
+                            _logger.TraceEvent(TraceEventType.Error, 0, $"[{badAnalyzerInfo.AnalyzerId} ({badAnalyzerInfo.AnalyzerIdHash})] LOF: {badAnalyzerInfo.LOF}, Avg: {badAnalyzerInfo.Average}, Stddev: {badAnalyzerInfo.AdjustedStandardDeviation}");
```

Comment:
:bulb: If `_logger` is normally disabled, it would be good to avoid the string interpolation execution/allocation in cases where the result won't be used. If this is called only rarely then it's not a problem like this.

Reply:
this is servicehub logger. logger will be always there.